### PR TITLE
wrap validateType in DEBUG flag

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/@ember/application/globals-resolver.js [hunk 1]
```diff
@@ -7,10 +7,34 @@ import { get, findNamespace } from '@ember/-internals/metal';
 import { assert, info } from '@ember/debug';
 import { capitalize, classify, dasherize, decamelize } from '@ember/string';
 import { Object as EmberObject } from '@ember/-internals/runtime';
-import validateType from './lib/validate-type';
 import { getTemplate } from '@ember/-internals/glimmer';
 import { DEBUG } from '@glimmer/env';
 
+function validateType(resolvedType, parsedName) {
```
Comment (rwjblue) [inline packages/@ember/application/globals-resolver.js]: This seems to only be used in the one location below, can we remove this function and inline the implementation down there instead? (allows us to strip out the function as well)

File: packages/@ember/application/globals-resolver.js [hunk 2]
```diff
@@ -144,10 +168,9 @@ class DefaultResolver extends EmberObject {
       if (parsedName.root && parsedName.root.LOG_RESOLVER) {
         this._logLookup(resolved, parsedName);
       }
-    }
-
-    if (resolved) {
-      validateType(resolved, parsedName);
+      if (resolved) {
+        validateType(resolved, parsedName);
+      }
```
Comment (rwjblue) [inline packages/@ember/application/globals-resolver.js]: Something like this:

```js
if (DEBUG && resolved) {
  let VALIDATED_TYPES = {
    route: ['isRouteFactory', 'Ember.Route'],
    component: ['isComponentFactory', 'Ember.Component'],
    view: ['isViewFactory', 'Ember.View'],
    service: ['isServiceFactory', 'Ember.Service'],
  };

  let validationAttributes = VALIDATED_TYPES[parsedName.type];

  if (validationAttributes) {
    let [factoryFlag, expectedType] = validationAttributes;

    assert(
      `Expected ${parsedName.fullName} to resolve to an ${expectedType} but ` +
        `instead it was ${resolvedType}.`,
      Boolean(resolvedType[factoryFlag])
    );
  } 
}
```

File: packages/@ember/application/globals-resolver.js (hunk 1 above)
Comment (bekzod) [reply packages/@ember/application/globals-resolver.js]: yeah I was going to do that at first, but then wanted to confirm if that is fine

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Restarted browser stack job.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**